### ISSUE TITLE
[On Hold] Improve error message when a debit card is added

### ISSUE THREAD

**Issue by marcochavezf** [Expensify/App#12913]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Break down in numbered steps

1. In NewDot sign in with an expensifail account
2. Click on the Avatar icon (Settings) -> Payments -> Add Payment method -> Debit card
3. Fill in the form and for the card number enter a valid non-debit [testing card](https://rftdevportal.thebancorp.com/reference/test-sandbox-cards) like `4111111111111111`

## Expected Result:
A descriptive error message should be shown. 

## Actual Result:
The error `Auth CreateFund returned an error` is shown. eg: conor@expensifail.com, getting [these logs](https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A(%2276de56176b6daade-SJC%22)%20AND%20timestamp%3A%5B2022-11-22T01%3A46%3A43.974Z%20TO%202022-11-22T03%3A46%3A43.974Z%5D&index=_all) with the error showing as `400 Invalid debit card`

## Workaround:

We could look up the logs for the user to find the error message and communicate that to the customer to solve

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 
**Reproducible in staging?:** 
**Reproducible in production?:** 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:** 
**Issue reported by:** 
**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<img width="798" alt="203069941-eba69df7-7444-4d02-9a9a-c76ebc67ca43" src="https://user-images.githubusercontent.com/6829422/203194239-a5cb90b4-61d1-4e61-bab8-46fc6b0669f3.png">

**Comment by conorpendergrast** [comment]
From this SO: https://stackoverflow.com/c/expensify/questions/14418
- [X] **The "bug" is actually a bug:** Yes, there's no copy for this error
- [X] **The bug is not a duplicate report of an existing GH (close the GH and add any novel details to the original GH instead)**: Correct
- [X] **The bug is reproducible, following the reproduction steps:** I reproduced through conor@expensifail.com, getting [these logs](https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A(%2276de56176b6daade-SJC%22)%20AND%20timestamp%3A%5B2022-11-22T01%3A46%3A43.974Z%20TO%202022-11-22T03%3A46%3A43.974Z%5D&index=_all) with the error showing as `400 Invalid debit card`
- [X] **If you’re unable to reproduce the bug, add the _Needs reproduction_ label. Comment on the issue outlining the steps you took to try to reproduce the bug, your results and tag the issue reporter and the Applause QA member who created the issue. Ask them to clarify reproduction steps and/or to check the reproduction steps again. Close issue.**: N/A, I could reproduce
- [X] **The GH template is filled out as fully as possible -- this means the GH body and title are clear (ie. could an external contributor understand it and work on it?):** I added logs and error message example, and updated the workaround 
- [X] **The GH was created by an Expensify employee or a QA tester:** Yep
- [X] **If the bug is an OldDot issue, you should decide whether it is widespread enough to justify fixing or it is better to wait for reunification. If it’s better to wait, close the GH & provide this justification:** N/A, it's New Dot
- [X] **If there's a link to Slack, check the discussion to see if we decided not to fix it:** no Slack link, and I checked the PR that mentioned this issue
- [X] **Decide if the GH should be resolved by an External contributor or Internal engineer, add the appropriate label:** Internal, and Marco has already added that label

**Comment by conorpendergrast** [comment]
Confirming Bug0 steps for Internal issues [via Slack](https://expensify.slack.com/archives/C01SKUP7QR0/p1669671639691139), and then will take those here!

**Comment by conorpendergrast** [comment]
Hah, @marcochavezf has [confirmed](https://expensify.slack.com/archives/C01SKUP7QR0/p1669674797366489?thread_ts=1669671639.691139&cid=C01SKUP7QR0) that he is working on a PR for the issue. Marco, can you link the draft PR to this issue?

**Comment by marcochavezf** [comment]
PR up to be reviewed, but on hold because of the [Bancorp routing configuration](https://expensify.slack.com/archives/C03TQ48KC/p1669674102027139).

**Comment by marcochavezf** [comment]
Hi @rosegrech, just gut-checking if the copy for the error message when a user enters a non-debit card is fine (`Make sure the entered debit card is a valid one`):

<img width="911" alt="Screenshot 2022-11-28 at 23 12 58" src="https://user-images.githubusercontent.com/6829422/204445164-b5113b73-4aa5-4af5-9864-4e88ea338509.png">

**Comment by rosegrech** [comment]
sure thing, this is a bit smoother:

> Make sure the debit card you entered is valid.

**Comment by conorpendergrast** [comment]
Do we have any idea what makes a debit card "valid" or "not valid"?

**Comment by marcochavezf** [comment]
> Do we have any idea what makes a debit card "valid" or "not valid"?

We only allow debit cards for this flow, and there's a possibility that the user could enter a credit or pre-paid card.

**Comment by conorpendergrast** [comment]
Ah ok, got it, so there's two possibilities: 
1. They've entered an invalid debit card
2. They've entered something that isn't debit card

> Make sure the debit card you entered is valid.

Seems to address 1 fine, but it's a little subtle on 2. @rosegrech what do you think on
> Make sure you've entered a valid debit card.

It's a very slight rewording, but should emphasise that it has to be valid, and a debit cared

**Comment by rosegrech** [comment]
Great detective work Conor!

**Comment by marcochavezf** [comment]
Cool, then `Make sure you've entered a valid debit card.` will be the final copy, thanks! 

Additionally, I'm going to put this one on hold until we are able to [communicate with Bancorp sandbox API](https://github.com/Expensify/Salt/pull/11676#issuecomment-1337156324) to use test debit cards on staging/production, otherwise, Applause won't be able to test the fix when the PR is deployed.

**Comment by conorpendergrast** [comment]
I'm re-assigning to another BugZero team member in case there are any further steps, as I'm going on sabbatical in 30 minutes!

**Comment by conorpendergrast** [comment]
@isabelastisser Nothing needed her for now, assigning you in case there's any follow-up questions from @marcochavez for anything else

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by isabelastisser** [comment]
Not overdue, changing this to weekly.

**Comment by marcochavezf** [comment]
Not overdue

**Comment by marcochavezf** [comment]
Still waiting on Bancorp's response. I'm going to use this [issue](https://github.com/Expensify/Expensify/issues/237181) to post updates about it.

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by isabelastisser** [comment]
Internal; not overdue.

**Comment by isabelastisser** [comment]
Internal, can I change this to `Monthly`? @marcochavezf

**Comment by marcochavezf** [comment]
Ah yeah, that's fine. I hope to continue with this one soon (updates [here](https://github.com/Expensify/Expensify/issues/237181#issuecomment-1419700519)).

**Comment by marcochavezf** [comment]
No updated, focused on EYOC MVP this week

**Comment by marcochavezf** [comment]
> No updated, focused on EYOC MVP this week

**Comment by marcochavezf** [comment]
> No updated, focused on EYOC MVP this week

**Comment by marcochavezf** [comment]
Still on hold

**Comment by marcochavezf** [comment]
Still on hold

**Comment by isabelastisser** [comment]
Not overdue, on hold!

**Comment by marcochavezf** [comment]
Removing the hold title since the debit test cards are not working on staging. I will work on this one in spare cycles

**Comment by slafortune** [comment]
👋 hey, quick note that we should keep all e/app GHs as daily unless they have "Hold" in the title. I'm going to flip this back to daily as I'm doing the resync and it's [step in the SO process guide](https://stackoverflowteams.com/c/expensify/questions/8208/14658#14658).

**Comment by isabelastisser** [comment]
Internal!

**Comment by isabelastisser** [comment]
Not overdue, internal!

**Comment by marcochavezf** [comment]
Not overdue, focused on unauthenticated users in public rooms implementation for RenderATL

**Comment by marcochavezf** [comment]
Not overdue, focused on unauthenticated users in public rooms implementation for RenderATL

**Comment by isabelastisser** [comment]
Hey @marcochavezf, any updates here?

**Comment by isabelastisser** [comment]
Not overdue.

**Comment by marcochavezf** [comment]
Not overdue.

**Comment by isabelastisser** [comment]
Not overdue, on hold!

**Comment by sophiepintoraetz** [comment]
@isabelastisser - for next time, it would be great to include a summary of what's happened and what needs to happen while you're away!

@marcochavezf - what issue are we holding on? I see two are referenced earlier that have merged/closed?

**Comment by sophiepintoraetz** [comment]
>I haven't had the chance to come back to it because other hight priorities and since this is feature is still behind the wallet beta. Since things have been slowed down since SHN, I have that on my list of priorities to tackle soon :)👍🏾1⠀⠀

From Marco!

**Comment by sophiepintoraetz** [comment]
@marcochavezf - because this is an old bug, I'm not sure we should downgrade the priority - see Puneet's post [here](https://expensify.slack.com/archives/C056AV5V8UX/p1687892112708039). Let me know if I can help with prioritising!

**Comment by sophiepintoraetz** [comment]
@marcochavezf - let me know when you can get to an update!

**Comment by marcochavezf** [comment]
Created a [PR](https://github.com/Expensify/Web-Expensify/pull/38044) to change the message, but the Bancorp sandbox token is [expired again](https://github.com/Expensify/Expensify/issues/243565#issuecomment-1620686070). I think we'd need to put it on hold until the token is updated.

**Comment by isabelastisser** [comment]
I'm back! Thanks, @sophiepintoraetz !

**Comment by mountiny** [comment]
Noting that it seems like the PR still did not fix the error message

**Comment by mountiny** [comment]
bumping here, I believe QA found this was not solve afaik @marcochavezf

**Comment by isabelastisser** [comment]
Not overdue!

**Comment by marcochavezf** [comment]
> bumping here, I believe QA found this was not solve afaik @marcochavezf

Ah yes, that's because of the Bancorp token expired again, waiting for the new token [here](https://github.com/Expensify/Expensify/issues/296548#issuecomment-1634654580)

**Comment by marcochavezf** [comment]
Still waiting for the new token

**Comment by isabelastisser** [comment]
Not overdue, on hold!

**Comment by marcochavezf** [comment]
Seems new [sandbox token](https://github.com/Expensify/Auth/pull/8332) is not working locally, I will retest again soon

**Comment by marcochavezf** [comment]
We would need to contact Bancorp again because the token is [expired](https://github.com/Expensify/Auth/pull/8332#issuecomment-1655813161) too. Moving it to weekly

**Comment by isabelastisser** [comment]
Still on hold! @marcochavezf, did you reach out to Bancorp to follow up on the token issue?

**Comment by isabelastisser** [comment]
Still on hold.

**Comment by isabelastisser** [comment]
Internal, still on hold.

**Comment by isabelastisser** [comment]
@marcochavezf any progress here? Thanks!

**Comment by marcochavezf** [comment]
We're depending on a new sandbox token but we're focused on waves

**Comment by isabelastisser** [comment]
Not overdue, on hold.

**Comment by marcochavezf** [comment]
This GH also depends on a new sandbox token, more details [here](https://github.com/Expensify/App/issues/12911#issuecomment-1745753223)

**Comment by isabelastisser** [comment]
on hold!

**Comment by isabelastisser** [comment]
Still on hold!

**Comment by isabelastisser** [comment]
Moving to  monthly, still on hold.

**Comment by isabelastisser** [comment]
@marcochavezf still on hold? Should we close this, given our new priorities?

**Comment by isabelastisser** [comment]
Bump @marcochavezf on my question above. Thanks!

**Comment by marcochavezf** [comment]
Yeah, sounds good, let's close this one
